### PR TITLE
Implementing "deducing this"

### PR DESCRIPTION
**Pull request by fpelliccioni**

### CODE REVIEW

File: src/lib/AST/ASTVisitor.cpp [hunk 1]
```diff
@@ -1800,6 +1800,7 @@ class ASTVisitor
                 convertToReferenceKind(
                     D->getRefQualifier());
             I.specs0.isFinal |= D->template hasAttr<FinalAttr>();
+            I.specs1.isExplicitObjectMemberFunction |= D->isExplicitObjectMemberFunction();
```
Comment (sdkrystian) [inline src/lib/AST/ASTVisitor.cpp:1803]: we may want to use `FunctionDecl::hasCXXFunctionObjectParameter` instead as `CXXMethodDecl::isExplicitObjectMemberFunction` has surprising behavior at times with respect to storage class specifiers.

File: share/mrdocs/addons/generator/asciidoc/partials/signature/function.adoc.hbs [hunk 2]
```diff
@@ -11,7 +11,7 @@
 {{#if (eq symbol.class "normal")}}{{>declarator-before symbol.return}}
 {{/if~}}
 {{>declarator-id symbol link=symbol.template.primary}}
-({{#each symbol.params}}{{#unless (and @first @last)}}
+({{#if symbol.isExplicitObjectMemberFunction}}this {{/if~}}{{#each symbol.params}}{{#unless (and @first @last)}}
```
Comment (sdkrystian) [inline share/mrdocs/addons/generator/asciidoc/partials/signature/function.adoc.hbs:14]: extraneous tilde after `/if`

**Review by sdkrystian — CHANGES_REQUESTED** [review]

File: share/mrdocs/addons/generator/asciidoc/partials/signature/function.adoc.hbs (hunk 2 above)
Comment (fpelliccioni) [reply share/mrdocs/addons/generator/asciidoc/partials/signature/function.adoc.hbs:14]:
```suggestion
({{#if symbol.isExplicitObjectMemberFunction}}this {{/if}}{{#each symbol.params}}{{#unless (and @first @last)}}
```

File: src/lib/AST/ASTVisitor.cpp (hunk 1 above)
Comment (fpelliccioni) [reply src/lib/AST/ASTVisitor.cpp:1803]: I added it here:

https://github.com/cppalliance/mrdocs/pull/541/files#diff-908f6a978cfcc9776770bcb54285299028b018608537c471f5ce7d78042a2f79R1788

Do you think I have to keep both or should I delete this line in which I use `CXXMethodDecl::isExplicitObjectMemberFunction`?

Comment (sdkrystian) [reply src/lib/AST/ASTVisitor.cpp:1803]: @fpelliccioni Sorry for the belated response
> Do you think I have to keep both or should I delete this line in which I use 
only keep the line that uses `hasCXXFunctionObjectParameter`

**Review by sdkrystian — CHANGES_REQUESTED** [review]
Please add tests in `test-files/old-tests`

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -35,7 +35,7 @@ jobs:
             gcc 13.1
             clang 17
             msvc 14.34
-          standards: '20'
+          standards: '23'
```
Comment (alandefreitas) [inline .github/workflows/ci.yml:38]: Not here

**Comment by fpelliccioni** [comment]
> Please add tests in `test-files/old-tests`. Otherwise, everything looks good.



> Please add tests in `test-files/old-tests`. Otherwise, everything looks good.

Done @sdkrystian

**Review by sdkrystian — APPROVED** [review]
LGTM, but please exclude the whitespace only changes to `ci.yml`

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -107,7 +107,7 @@ jobs:
               echo "Neither curl nor wget are available"
               exit 1
           fi
-          
+
```
Comment (fpelliccioni) [inline .github/workflows/ci.yml:110]:
```suggestion
          
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -95,7 +95,7 @@ jobs:
           config_type="Release"
           filename="${{ runner.os }}-$config_type-29b20829.${{ ( runner.os == 'Windows' && '7z' ) || 'tar.xz' }}"
           url="https://mrdox.com/llvm+clang/$filename"
-          
+
```
Comment (fpelliccioni) [inline .github/workflows/ci.yml:98]:
```suggestion
          
```

**Pull request merged**